pkg/logs/launchers/file/launcher.go

```diff
@@ -56,6 +57,10 @@ type Launcher struct {
 	scanPeriod             time.Duration
 	flarecontroller        *flareController.FlareController
 	tagger                 tagger.Component
+	filesChan              chan []*tailer.File
+	addedSourceTailers     []*tailer.File
+	ctx                    context.Context
```

Comment:
Since `ctx` and `cancel` are onyl ever used in the run loop, can we move these into function local variables inside `run()`? this way we dont carry additioal state on the launcher struct (since we don't need it anywhere else)

Reply:
Yep, changing that now